Makefile

```diff
@@ -99,19 +99,13 @@ sdist_name:=toil-$(dist_version).tar.gz
 docker_tag:=$(shell $(python) version_template.py dockerTag)
 default_docker_registry:=$(shell $(python) version_template.py dockerRegistry)
 docker_path:=$(strip $(shell which docker))
-ifdef docker_path
```

Comment:
Why was this check removed? It seems necessary.

Reply:
This checks if docker is installed for the `@needs_appliance` decorator to disable docker tests.  But if you look at the def, it already does that in `def needs_appliance`.  So first of all, it's redundant and unnecessary.

Second, its purpose is to set `TOIL_APPLIANCE_SELF` to `None` if docker is not installed.  Since this checks that `TOIL_APPLIANCE_SELF` is valid, this is painfully unhelpful.